src/Arcus.Messaging.Abstractions.EventHubs/Extensions/EventDataExtensions.cs

```diff
@@ -0,0 +1,73 @@
+        /// </summary>
+        /// <param name="message">The consumed Azure EventHubs event to describe the messaging context.</param>
+        /// <param name="eventHubsNamespace">
+        ///     The fully qualified Event Hubs namespace that the processor is associated with.
+        ///     This is likely to be similar to <c>{yournamespace}.servicebus.windows.net</c>.
+        /// </param>
+        /// <param name="eventHubsName"> The name of the Event Hub that the processor is connected to, specific to the EventHubs namespace that contains it.</param>
+        /// <exception cref="ArgumentNullException">Thrown when the <paramref name="message"/> is <c>null</c>.</exception>
+        /// <exception cref="ArgumentException">
+        ///     Thrown when the <paramref name="eventHubsNamespace"/>, or <paramref name="eventHubsName"/> is blank.
+        /// </exception>
+        public static AzureEventHubsMessageContext GetMessageContext(this EventData message, string eventHubsNamespace, string eventHubsName)
+        {
+            Guard.NotNull(message, nameof(message), "Requires an event data to retrieve the Azure EventHubs message context");
+            Guard.NotNullOrWhitespace(eventHubsNamespace, nameof(eventHubsNamespace), "Requires an Azure EventHubs namespace to built-up the message context");
+            Guard.NotNullOrWhitespace(eventHubsName, nameof(eventHubsName), "Requires an Azure EventHubs name to built-up the message context");
+
+            return GetMessageContext(message, eventHubsNamespace, eventHubsName, "$Default");
+        }
+
+        /// <summary>
+        /// Gets the message contextual information from the received <paramref name="message"/>.
+        /// </summary>
+        /// <param name="message">The consumed Azure EventHubs event to describe the messaging context.</param>
+        /// <param name="eventHubsNamespace">
+        ///     The fully qualified Event Hubs namespace that the processor is associated with.
+        ///     This is likely to be similar to <c>{yournamespace}.servicebus.windows.net</c>.
+        /// </param>
+        /// <param name="eventHubsName"> The name of the Event Hub that the processor is connected to, specific to the EventHubs namespace that contains it.</param>
+        /// <param name="consumerGroup">The name of the consumer group this event processor is associated with.</param>
+        /// <exception cref="ArgumentNullException">Thrown when the <paramref name="message"/> is <c>null</c>.</exception>
+        /// <exception cref="ArgumentException">
+        ///     Thrown when the <paramref name="eventHubsNamespace"/>, <paramref name="consumerGroup"/>, or <paramref name="eventHubsName"/> is blank.
+        /// </exception>
+        public static AzureEventHubsMessageContext GetMessageContext(this EventData message, string eventHubsNamespace, string eventHubsName, string consumerGroup)
```

Comment:
I'm just wondering if the eventHubNamespace, eventHubName and consumerGroup information is not present in the systemProperties of the EventData instance.   If that is the case, we might not need to pass in those 3.

Reply:
Tried to look for it, but like in Service Bus scenarios, this wasn't there. https://learn.microsoft.com/en-us/dotnet/api/microsoft.azure.eventhubs.eventdata.systempropertiescollection?view=azure-dotnet

Reply:
Indeed, I've also inspected some real events on an eventhub, but that info is not available.